src/IssueViz/Models/AnalysisIssueVisualizationConverter.cs

```diff
@@ -123,8 +123,9 @@ private IReadOnlyList<IQuickFixVisualization> GetQuickFixVisualizations(IAnalysi
                 var editVisualizations = fix.Edits.Select(edit =>
                 {
                     var editSpan = issueSpanCalculator.CalculateSpan(edit, textSnapshot);
+                    var originalText = textSnapshot.GetText(editSpan);
```

Comment:
This is the first computation: when we calculate the original span, we also store a copy of the text as it was during the calculation. This code will be called immediately upon getting the analysis results and (hopefully) will be identical to what was analyzed.
It is still possible that this would be wrong: for example, if we trigger an analysis and it takes 1 minute to return back the issue from the subprocess. During this time, the user might have edited the code, and both the span and the text would be incorrect.
I don't think we have a way to handle that, not unless the subprocess also returns us the text that they used to compute the fix.

Reply:
> It is still possible that this would be wrong: for example, if we trigger an analysis and it takes 1 minute to return back the issue from the subprocess.

We already try to minimise this risk as much as possible. The `AccumulatingIssueConsumer` has a reference to the snapshot at the point the analysis was triggered. That is the snapshot that is passed to the issue converter, so any subsequent edits by the user won't be visible.

There is a window in which the user could make edits that would mess things up, but that window is between when we trigger the analysis and the subprocess reads in the file.

There's a comment about this in the `AccumulatingIssueConsumer`. Maybe we need a short comment in the converter too e.g. that the caller is responsible for providing the snapshot that matches the raised issues.


https://github.com/SonarSource/sonarlint-visualstudio/blob/63a9400ef82f8e62298b8ab2eb433723dc17a8f8/src/Integration.Vsix/SonarLintTagger/AccumulatingIssueConsumer.cs#L48